### PR TITLE
mock endepunkt for manuell opprettelse av meldekort

### PR DESCRIPTION
**Pull request by christinaroise**
## Beskrivelse

  Implementerer mock-endepunkter for manuell opprettelse av meldekort. Backend for denne funksjonaliteten eksisterer ikke enda, så dette er en frontend-mock for å kunne videreutvikle UI-et.

  ## Endringer

  ### Nye felt
  - Lagt til `opprettetManuelt?: boolean | null` i `IRapporteringsperiode` type
    - Optional-felt som indikerer om et meldekort ble opprettet manuelt av saksbehandler
    - Oppdatert database schema tilsvarende

  ### Nye funksjoner i `app/mocks/db.ts`

  **`beregnAntallMeldekort(fraDato: string, tilDato: string): number`**
  - Beregner antall komplette 2-ukers perioder i et datoområde
  - Starter fra første mandag på eller etter startdato
  - En **sterk forenkling** av hvordan vi beregner meldekortperioder

  **`opprettManueltMeldekort(fraDato: string, tilDato: string, ident: string, kilde: IKilde): IRapporteringsperiode[]`**
  - Oppretter meldekort for alle 2-ukers perioder i et datoområde
  - Setter riktig meldekorttype:
    - `ETTERREGISTRERT` for perioder i fortiden
    - `ORDINAERT` for fremtidige perioder
  - Sjekker for duplikater og hopper over eksisterende perioder
  - Setter `opprettetManuelt: true` på alle opprettede meldekort
  - Bruker eksisterende helpers (`lagDager()`, `lagRapporteringsperiode()`)

  ### Nye mock-endepunkter i `app/mocks/mock-meldekortregister.ts`

  **GET `/sb/person/:personId/meldekort/tell-meldekort`**
  - Query params: `fraDato`, `tilDato`
  - Returnerer: `{ antall: number }`
  - Beregner hvor mange meldekort som vil bli opprettet i datoområdet

  **POST `/sb/person/:personId/meldekort/opprett-manuelt`**
  - Request body: `{ fraDato: string, tilDato: string, kilde: IKilde }`
  - Returnerer: `{ antall: number, meldekortIds: string[] }`
  - Oppretter meldekort og returnerer antall og ID-er

  ### Tester
  - 20 nye tester i `app/mocks/db.test.ts`:
    - 6 tester for `beregnAntallMeldekort()`
    - 14 tester for `opprettManueltMeldekort()`
       - Dekker edge cases som duplikater, ulike personer, datotyper, etc.

### CODE REVIEW

File: app/utils/types.ts [hunk 1]
```diff
@@ -125,6 +125,8 @@ export interface IRapporteringsperiode {
   /** yyyy-mm-dd, dato for når meldekortet skal regnes som registrert, kan settes av saksbehandler ellers følger det av innsendtTidspunkt */
   meldedato: string | null;
   registrertArbeidssoker: boolean | null;
+  /** Indikerer om meldekortet ble opprettet manuelt av saksbehandler */
+  opprettetManuelt?: boolean | null;
```
Comment (vskjefst) [inline app/utils/types.ts]: Hadde vurdert å endre denne til `opprettetManueltAvSaksbehandler`. Da kan kommentaren fjernes, og det er ingen tvil om hvordan dette meldekortet har oppstått når feltet brukes andre steder i koden heller.

**Review by vskjefst — APPROVED** [review]
